utils/wasm-gen/src/config.rs

```diff
@@ -131,6 +131,11 @@ impl ConfigsBundle for (GearWasmGeneratorConfig, SelectableParams) {
 pub struct StandardGearWasmConfigsBundle {
     /// Externalities to be logged.
     pub log_info: Option<String>,
+    /// Frequency of wait syscalls.
+    ///
+    /// For example, if this parameter is 4, wait syscalls will be invoked
+    /// with probability 1/4.
+    pub waiting_frequency: Option<u32>,
```

Comment:
I'm not sure that **frequency** is a suitable name for that.
You see, `SyscallInjectionType::Function` actually defines **frequency**.

Reply:
I have here a suggestion to make this solution more general, which will make `wasm-gen` even better.

What about inserting not only the frequency, but the chance to be called in the `SyscallInjectionType::Function`?
So it will be `SyscallInjectionType::Function { call_chance: Option<u32>, frequency: RangeInclusive<u32> }`.

In this case when generating instructions for any syscall we will first process `call_chance`. If it's `None`, then no need in defining instructions like `if syscall_called_ptr & call_chance ...`.

The only downside here is that we must define a layout in memory which stores flags for every syscall, like it's done here for `wait`. Maybe there are some ways to avoid that gracefully.

All in all, that will make handling `wait` syscall invocation not that alien to `build_syscall_invoke_instructions` and the whole design.